core/src/main/java/org/apache/iceberg/view/ViewMetadata.java

```diff
@@ -91,41 +111,325 @@ default ViewMetadata checkAndNormalize() {
         "Unsupported format version: %s",
         formatVersion());
 
-    Preconditions.checkArgument(versions().size() > 0, "Invalid view versions: empty");
-    Preconditions.checkArgument(history().size() > 0, "Invalid view history: empty");
-    Preconditions.checkArgument(schemas().size() > 0, "Invalid schemas: empty");
+    return this;
```

Comment:
Why does this method return `this`? Other check methods are defined as `default void check() { ... }`

Reply:
this was because we initially did history **normalization** in this method, which required to return either `this` or a new instance

Reply:
Thanks for catching this! I missed it in my PR.